cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginConfig.java

```diff
@@ -37,4 +40,23 @@
   public final PluginProperties getProperties() {
     return properties;
   }
+
+  /**
+   * Returns true if property value is a macro; false otherwise.
+   * @param fieldName
+   * @return
+   */
+  public boolean isMacro(String fieldName) {
+    try {
+      Annotation annotation = this.getClass().getDeclaredField(fieldName).getAnnotation(Macro.class);
+      if (annotation == null) {
+        return false;
+      } else {
+        // field is macro enabled, now check if field value is a macro
+        return new MacroUtils().isMacro(properties.getProperties().get(fieldName));
```

Comment:
Although the `MacroUtils` is going to go away (as @albertshau mentioned above), we should never "new" a util class.